cdap-ui/app/cdap/components/Replicator/Detail/ThroughputGraph/index.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState, useEffect, useContext } from 'react';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
+import { MyMetricApi } from 'api/metric';
+import {
+  COLOR_MAP,
+  IThroughputData,
+  renderThroughputGraph,
+} from 'components/Replicator/Detail/ThroughputGraph/throughput';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import MetricsQueryHelper from 'services/MetricsQueryHelper';
+import { PROGRAM_INFO } from 'components/Replicator/constants';
+import { DetailContext } from 'components/Replicator/Detail';
+import { throughputParser } from 'components/Replicator/Detail/ThroughputGraph/parser';
+import { Observable } from 'rxjs/Observable';
+import Heading, { HeadingTypes } from 'components/Heading';
+
+const styles = (): StyleRules => {
+  return {
+    root: {
+      position: 'relative',
+    },
+    heading: {
+      textAlign: 'center',
+    },
+    square: {
+      display: 'inline-block',
+      height: '15px',
+      width: '15px',
+      marginRight: '7px',
+    },
+    inserts: {
+      backgroundColor: COLOR_MAP.inserts,
+    },
+    updates: {
+      backgroundColor: COLOR_MAP.updates,
+    },
+    deletes: {
+      backgroundColor: COLOR_MAP.deletes,
+    },
+    bottomLegend: {
+      marginTop: '25px',
+      padding: '0 15px',
+      display: 'flex',
+
+      '& > div': {
+        marginRight: '15px',
+      },
+    },
+  };
+};
+
+const CONTAINER_ID = 'replication-throughput-graph';
+
+const ThroughputGraphView: React.FC<WithStyles<typeof styles>> = ({ classes }) => {
```

Comment:
Do you think we'll use this pattern for rendering D3 graphs? Should we make some kind of higher-order component to handle the common features like resizing?

Reply:
Maybe HOC is the wrong idea... it could be injectable like
```
<ChartContainer
  data={thoughputData}
  chartRenderFunc={throughputChart}
  width={w}
  height={h}
  ...
/>
```

or even

```
<BarChart
  data={throughputData}
  traces=['inserts', 'updates', 'deletes', 'errors']
  ...
/>
```

Reply:
I like the idea of ChartContainer that will handle the window resizing. For specific BarCharts, so far our graphs had been too customized that I would be weary about abstracting it out. I think I will start with the ChartContainer, then moving forward if we observe another pattern for specific graph, we can add another layer of abstraction